pkg/admission/validator/lakom/shoot.go

```diff
@@ -44,16 +50,78 @@ func findExtension(extensions []core.Extension, extensionType string) (int, core
 
 func (s *shoot) validateScopeType(fldPath *field.Path, scopeType lakom.ScopeType) field.ErrorList {
 	errList := field.ErrorList{}
-
 	if !lakom.AllowedScopes.Has(scopeType) {
 		errList = append(errList, field.NotSupported(fldPath, scopeType, lakom.AllowedScopes.UnsortedList()))
 	}
 
 	return errList
 }
 
+// TODO: This check exists in the validation of the lakom config as well. It can be extracted as a util function
+func (s *shoot) validateCosignPublicKeys(fldPath *field.Path, cosignPublicKeys []config.Key) field.ErrorList {
+	errList := field.ErrorList{}
+
+	usedNames := map[string]any{}
+	for idx, k := range cosignPublicKeys {
+		if k.Name == "" {
+			errList = append(errList, field.Required(fldPath.Index(idx), "key name should no be empty"))
+		}
+
+		if _, ok := usedNames[k.Name]; ok {
+			errList = append(errList, field.Duplicate(fldPath.Index(idx), k.Name))
+		}
+		usedNames[k.Name] = nil
```

Comment:
I think this code should be skipped when the name is empty, what do you think?

Reply:
Yeah, absolutely agree